src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,146 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private int arraySize;
+    private Object[] elementData;
+    private int size;
+
+    public ArrayList() {
+        arraySize = 10;
+        elementData = new Object[arraySize];
+        size = 0;
+    }
```

Comment:
Don't begin class or method implementation with an empty line. Also, any magic numbers should be constants. Consider declaring a constant for the initial array size instead of using the hardcoded value '10'.